src/request/reducers/RequestDetailDataReducer.ts

```diff
@@ -0,0 +1,174 @@
+import { IRequestDetailDataOperationState } from '../stores/IRequestDetailDataOperationState';
+import { selectOperationAction } from '../actions/RequestDetailDataActions';
+import { requestDetailUpdateAction } from '../actions/RequestDetailActions';
+
+import { ICommandAfterExecutePayload } from '../messages/ICommandAfterExecutePayload';
+import { ICommandBeforeExecutePayload } from '../messages/ICommandBeforeExecutePayload';
+import { IDataMongoDbDeletePayload } from '../messages/IDataMongoDbDeletePayload';
+import { IDataMongoDbInsertPayload } from '../messages/IDataMongoDbInsertPayload';
+import { IDataMongoDbReadPayload } from '../messages/IDataMongoDbReadPayload';
+import { IDataMongoDbUpdatePayload } from '../messages/IDataMongoDbUpdatePayload';
+import { IMessage, IMessageEnvelope } from '../messages/IMessageEnvelope';
+
+import { Action, combineReducers } from 'redux';
+
+import * as _ from 'lodash';
+
+interface ISortableOperation {
+    ordinal: number;
+    operation: IRequestDetailDataOperationState;
+}
+
+function updateSelectedIndex(state: number, request) {
+    return request
+        ? state
+        : 0; 
+}
```

Comment:
No just habit, as I find it easier to read the test/true/false cases on separate lines, in particular, when any of them are longer statements (such as HTML elements in TSX files).